src/core/reducers/autocomplete.js

```diff
@@ -0,0 +1,69 @@
+export const AUTOCOMPLETE_LOADED = 'core/autocomplete/AUTOCOMPLETE_LOADED';
+export const AUTOCOMPLETE_STARTED = 'core/autocomplete/AUTOCOMPLETE_STARTED';
+export const AUTOCOMPLETE_CANCELLED = 'core/autocomplete/AUTOCOMPLETE_CANCELLED';
+
+const initialState = {
+  loading: false,
+  suggestions: [],
+};
+
+export function autocompleteStart({ errorHandlerId, filters }) {
+  if (!errorHandlerId) {
+    throw new Error('errorHandlerId is required');
+  }
+
+  if (!filters) {
+    throw new Error('filters are required');
+  }
+
+  return {
+    type: AUTOCOMPLETE_STARTED,
+    payload: { errorHandlerId, filters },
+  };
+}
+
+export function autocompleteCancel() {
+  return { type: AUTOCOMPLETE_CANCELLED };
+}
+
+export function autocompleteLoad({ results }) {
+  if (!results) {
+    throw new Error('results are required');
+  }
+
+  return {
+    type: AUTOCOMPLETE_LOADED,
+    payload: { results },
+  };
+}
+
+export default function reducer(state = initialState, action = {}) {
+  const { payload } = action;
+
+  switch (action.type) {
+    case AUTOCOMPLETE_CANCELLED:
+      return {
+        ...state,
+        loading: false,
+        suggestions: [],
+      };
+    case AUTOCOMPLETE_STARTED:
+      return {
+        ...state,
+        loading: true,
+        suggestions: [],
+      };
+    case AUTOCOMPLETE_LOADED:
+      return {
+        ...state,
+        loading: false,
+        suggestions: payload.results
+        // TODO: Remove this when `null` names are not returned. See:
+        // https://github.com/mozilla/addons-server/issues/6189
+          .filter((result) => result.name !== null)
+          .map((result) => result.name),
```

Comment:
I am a bit reluctant to expose too many things in the state.

At first, I needed names only because that's what I had to display. In the UI PR, I have added the `url` and `iconUrl` because that's what I needed to match the new spec.